### DISCUSSION TITLE
Immediate Redirect to Provider

### DISCUSSION THREAD

**Discussion by jkupcho** [nextauthjs/next-auth#1435] [Help] [answered]
I'm looking at Keycloak to maintain external social logins, but I'm interested in using this library to take care of the OAuth handling for me as it can get tricky and I'd rather not roll that myself.

My question is, is there a way to immediately redirect to a provider without having the extra step of clicking a button to do the log in?

Thanks!

**Comment by balazsorban44** [comment, marked as answer]
There is, please read the docs :slightly_smiling_face: https://next-auth.js.org/getting-started/client#starts-google-oauth-sign-in-flow-when-clicked

(You replace `"google"` with the id of your provider)

**Reply by jkupcho** [reply]
You still need to click the button it seems, no? So, are you trying to say on page load, just do `signIn('provider_name')` without the button, essentially?

**Reply by balazsorban44** [reply]
I am not sure I understand. Maybe you want to create a page that requires auth?

I think I provide a solution for this here: https://github.com/nextauthjs/next-auth/issues/1210#issuecomment-782630909

But to simplify it, yes, you COULD start the login process in a `useEffect()`, just make sure you properly handle it so you don't end up in a login loop.

**Reply by jkupcho** [reply]
I just want to immediately force login via the provider on any page. So, I think the `useEffect()` approach would probably be the the best.

**Reply by jkupcho** [reply]
I tried out the solution you linked, worked perfectly, thanks!
